caf: power_manager: Document Zephyr PM features integration (#21917)
nrfconnect/sdk-nrf: doc/nrf/libraries/caf/power_manager.rst

Implied features
================

> The |power_manager| controls behavior of application modules to make the modules reduce power consumption when device is not used for a longer period of time.

Reviewer: ```suggestion
The |power_manager| controls the behavior of application modules to make the modules reduce power consumption when the device is inactive for a longer period of time.
```

> The :kconfig:option:`CONFIG_CAF_POWER_MANAGER` option also implies the following Zephyr features that can be used to reduce power consumption:

Reviewer: ```suggestion
The :kconfig:option:`CONFIG_CAF_POWER_MANAGER` Kconfig option also implies the following Zephyr features that can be used to reduce power consumption:
```

* System power off support (:kconfig:option:`CONFIG_POWEROFF`).
  Entering the system power off state reduces the power consumption further.
  The option is not implied for an nRF54H Series SoC (:kconfig:option:`CONFIG_SOC_SERIES_NRF54HX`), because the :c:func:`sys_poweroff` API is not yet fully supported on the nRF54H Series SoC.
* Device Power Management (:kconfig:option:`CONFIG_PM_DEVICE`).
  The option allows to reduce the power consumption of device drivers while they are inactive.

>   An application module that interacts with a driver supporting device power management APIs can use the APIs to control power state of the driver.

Reviewer: ```suggestion
  An application module that interacts with a driver supporting the device power management APIs can control the power state of the driver.
```

* Zephyr's :ref:`zephyr:pm-system` (:kconfig:option:`CONFIG_PM`).
  The nRF54H Series SoC (:kconfig:option:`CONFIG_SOC_SERIES_NRF54HX`) integrates the system power management to reduce power consumption when inactive.

>   When the |power_manager| switches application modules to low power mode, the CPU workload is lowered.

Reviewer: ```suggestion
  When the |power_manager| switches the application modules to low power mode, the CPU workload is lowered.
```
